packages/lodestar/src/sync/utils/sync.ts

```diff
@@ -105,7 +105,17 @@ export function fetchBlockChunks(
           return;
         }
         try {
-          yield await getBlockRange(logger, reqResp, peers, slotRange);
+          // a work around of timeout issue that cause our sync stall
+          let timer: NodeJS.Timeout | null = null;
+          yield (await Promise.race([
+            getBlockRange(logger, reqResp, peers, slotRange),
+            new Promise((_, reject) => {
+              timer = setTimeout(() => {
+                reject(new Error("beacon_blocks_by_range timeout"));
+              }, 3 * 60 * 1000); // 3 minutes
```

Comment:
where does the number of 3 minutes come from?

Reply:
in initial sync, we divide 64 slot range to a couple of small range and query it to eligible peers i parallel. Normally, it's pretty fast (only 2s - 10s) as I can see but if a chunk is failed, it'll try the same chunk with another peer.

In the case of timeout and an error is not thrown, it'll never finish the call and initial sync get stalled, the 3 minutes param is the maximum time we can wait and that's my estimate in the case. This rarely happens but quite annoying.

Reply:
make it a constant